packages/components/src/dateInput/DateInput.js

```diff
@@ -67,13 +65,15 @@ export default class DateInput extends React.Component<Props, State> {
     return state;
   }
 
-  componentDidUpdate(prevProps: Props) {
-    const { isDateRequested, onDateChange } = this.props;
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    const { onDateChange } = this.props;
     const { day, month, year } = this.state;
-    if (isDateRequested !== prevProps.isDateRequested && isDateRequested) {
-      if (day !== '' && year !== '') {
+    if (day !== prevState.day || year !== prevState.year) {
+      if (day.length === 2 && year.length === 4) {
```

Comment:
We had constants for that? `MAX_DAY_LENGTH`, etc...

Reply:
Yes we have, forgot about them :(

Reply:
fixed